cornflow-server/cornflow/commands/permissions.py

```diff
@@ -31,28 +50,70 @@ def register_base_permissions_command(external_app: str = None, verbose: bool =
         (perm.role_id, perm.action_id, perm.api_view_id) for perm in permissions_in_db
     ]
     resources_names = [resource["endpoint"] for resource in resources_to_register]
+    resources_roles_with_access = {
+        resource["endpoint"]: resource["resource"].ROLES_WITH_ACCESS
+        for resource in resources_to_register
+    }
+    # We extract a unique list of roles with access
+    roles_with_access = list(
+        set([role for roles in resources_roles_with_access.values() for role in roles])
+    )
+    roles_in_extra_permissions = [role for role, _, _ in extra_permissions]
+    roles_with_access = list(set(roles_with_access + roles_in_extra_permissions))
+    # We check if there is any role additional to the ones defined in the base permissions
+    additional_roles_with_access = [
+        role for role in roles_with_access if role not in ALL_DEFAULT_ROLES
+    ]
 
-    # Create base permissions
+    # We extract the existing roles in the database
+    existing_roles = [role.id for role in RoleModel.get_all_objects()]
+    new_roles_to_add = []
+    if len(additional_roles_with_access) > 0:
+        # If our id is not in the existing roles, we add it, applying a
+        # pre-defined name custom_role_<id>
+        for custom_role in additional_roles_with_access:
+            if custom_role not in existing_roles:
+                new_role = RoleModel(
+                    {
+                        "id": custom_role,
+                        "name": f"custom_role_{custom_role}",
```

Comment:
For now I think this is fine, but in the long run we probably should find a way to get the names and IDs of the extra roles directly from the external app